design-documents/service-isolation/invoker.md

```diff
@@ -0,0 +1,122 @@
+### Invoker API and SPI
+Responsible for contacting remote services
+#### API
+##### Remote call response
+Result of executing the call.
+Can be used by 3rd party devs to add data to remote calls results or read it when they choose to modify
+remote calling mechanism.
+```php
+interface InvokeResponseInterface {
+    /**
+     * @return mixed Value that remote service returned.
+     */
+    public function getResult();
+    
+    /**
+     * Exception that remote service has thrown, if any.
+     *
+     * @return \Throwable|null
+     */
+    public function getException(): ?\Throwable
+}
+```
+##### Remote Invoker
+Used to make a call to a Magento server.
+Can be used by 3rd party devs to write proxy services and modify invoker functionality/gateway.
+```php
+interface InvokerInterface {
+    /**
+     * @param string $class
+     * @param string $method
+     * @param array $arguments Keys - argument names.
+     * @return PromiseInterface Will provide InvokeResponseInterface when resolved,
+     *                 RemoteInvokeException on reject when networking fails,
+     *                 has "wait" method to use in synchronous way.
+     */
+    public function invoke(
+        string $class,
+        string $method,
+        array $arguments
+     ): PromiseInterface;
+}
+```
+ 
+#### SPI
+##### Base URL information
+Nodes data.
+Used to access nodes and provide requests for them.
+```php
+interface ServiceSourceInterface {
```

Comment:
I think we discussed, but it would be good to provide more description why do we need this and ServiceSourceRepositoryInteface interfaces.